homeassistant/components/vicare/entity.py

```diff
@@ -27,7 +35,42 @@ def __init__(
         if hasattr(device, "id"):
             self._attr_unique_id += f"-{device.id}"
 
-        self._attr_device_info = DeviceInfo(
+        if isinstance(device, HeatingCircuit):
+            self._attr_device_info = self._get_info_for_component(
+                device_config, device, "Circuit"
+            )
+        elif isinstance(device, FuelCellBurner | GazBurner | OilBurner):
+            self._attr_device_info = self._get_info_for_component(
+                device_config, device, "Burner"
+            )
+        elif isinstance(device, Compressor):
+            self._attr_device_info = self._get_info_for_component(
+                device_config, device, "Compressor"
+            )
+        else:
+            self._attr_device_info = self._get_info_for_device(device_config)
+
+    def _get_info_for_component(
+        self,
+        device_config: PyViCareDeviceConfig,
+        device: PyViCareDevice,
+        component_type: str,
+    ) -> DeviceInfo:
+        return DeviceInfo(
+            via_device=(DOMAIN, device_config.getConfig().serial),
+            identifiers={
+                (
+                    DOMAIN,
+                    f"{device_config.getConfig().serial}-{component_type.lower()}-{device.id}",
+                )
+            },
+            name=f"{component_type} {device.id + 1}",
```

Comment:
I don't like this solution. I've a PR pending that exposes the circuit name (https://github.com/somm15/PyViCare/pull/340) that can later on be used here for circuits.

Reply:
I agree, we should try to solve this upstream.
Not sure why we're appending a random incremental number anyhow to these? They devices doesn't have proper names?

Reply:
The heating device can have one or multiple burners / compressors and one or multiple heating circuits depending on the model. Each of these *components* have sensor values in the api (like circuit temp, operating hours, ..), so similar entities are created multiple times. 
Currently they are all assigned to the heating device itself and to distinguish them entity names get the component id (0, 1, 2, ...) appended to the name. Thus I currently cannot implements translation keys.

> Not sure why we're appending a random incremental number anyhow to these?

This I did to not list "burner 0" but "burner 1" as the first device.

Reply:
I meant that you are appending the number since I assume these actually don't have proper names provided by the api?
Given the user can't diff between burners and compressors anyhow why not just name them without the number?

Reply:
Changed to use simple names "Compressor" and "Burner".